### PR TITLE
New package: @ramseyinhouse/stylelint-config

### PR DESCRIPTION
**Pull request by ChristianJHughes**
## Description of Proposed Changes

Adds our stylelint configuration into a scoped `@ramseyinhouse/stylelint-config` package. It includes:

- `stylelint-config-sass-guidelines`: The base from which our `sass` guidelines are derived.
- `stylelint-order`: A plugin with which to enforce our concentric css standard.
- `stylelint-config-prettier`: Disables all of the rules that conflict with `prettier`, which is expected to work in tandem with this config.

When used with an IDE that runs `stylelint` and `prettier` on save, for example, CSS is formatted flawlessly. No additional IDE settings or weird overrides required.

## Testing Steps

I've verified that this package operates as expected when used within a local project.

I've also verified that the extends property within consuming configuration MUST use the fully qualified package name.

```
...
extends: ['@ramseyinhouse/stylelint-config']
...
```

### CODE REVIEW

File: packages/stylelint-config/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/RamseyInHouse/frontend.git"
+  },
+  "keywords": [
+    "ramsey",
+    "ramsey solutions",
+    "ramsey in house",
+    "stylelint",
+    "styelint-config"
```
Comment (alexmacarthur) [inline packages/stylelint-config/package.json:17]: 🚫  Tiny typo:

```suggestion
    "stylelint-config"
```

File: packages/stylelint-config/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  "version": "0.0.0",
+  "author": "Ramsey Solutions",
+  "description": "Ramsey Solutions standard stylelint configuration.",
+  "homepage": "https://github.com/ramseyinhouse/frontend/packages/stylelint-config",
+  "license": "GPL-3.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/RamseyInHouse/frontend.git"
+  },
+  "keywords": [
```
Comment (alexmacarthur) [inline packages/stylelint-config/package.json:12]: 🗣️  Maybe for better npm SEO?

```suggestion
  "keywords": [
     "css", 
     "scss",
     "sass",
```

**Review by alexmacarthur — COMMENTED** [review]
Nice work! A couple of comments/suggestions, including one blocker in the form of a typo.

File: packages/stylelint-config/lib/selectorClassPattern.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const selectorClassPatternRules = {
+  components:
+    '^([a-z]+-)?([A-Z][a-zA-Z0-9]+)(-[a-zA-Z0-9]+)?(--[a-zA-Z0-9]+)?$',
+  utilities:
+    '^u-((|palmOnly|lapOnly|lapToLapWide|lapToDesk|lapAndUp|lapWideOnly|lapWideToDesk|lapWideAndUp|deskOnly|deskAndUp|wallOnly)-)?[a-z][a-zA-Z0-9]+$',
```
[the remark this answers is not in the document]
Comment (ChristianJHughes) [reply packages/stylelint-config/lib/selectorClassPattern.js:5]: @alexmacarthur I think I agree -- let's connect on what our philosophy will be generally for proprietary features.

[the remark this answers is not in the document]
Comment (joshwhatk) [reply packages/stylelint-config/lib/selectorClassPattern.js:5]: @ChristianJHughes, @alexmacarthur, ooh! Sounds like an ADR may be in order 😁

I would LOVE to have our use of ADRs reflected in our open source packages!!

[the remark this answers is not in the document]
Comment (alexmacarthur) [reply packages/stylelint-config/lib/selectorClassPattern.js:5]: @joshwhatk oh my goodness YES

[the remark this answers is not in the document]
Comment (alexmacarthur) [reply packages/stylelint-config/lib/selectorClassPattern.js:5]: For the time being, we're gonna leave the pattern as it is. Once our configurations are all migrated here, we can have conversations about how we want to improve/iterate on it.

**Review by alexmacarthur — APPROVED** [review]

**Pull request merged**